pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -0,0 +1,393 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+	"log"
+	"time"
+)
+
+func resourcePagerDutyEventOrchestrationPathRouter() *schema.Resource {
+	return &schema.Resource{
+		Read:   resourcePagerDutyEventOrchestrationPathRouterRead,
+		Create: resourcePagerDutyEventOrchestrationPathRouterCreate,
+		Update: resourcePagerDutyEventOrchestrationPathRouterUpdate,
+		Delete: resourcePagerDutyEventOrchestrationPathRouterDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourcePagerDutyEventOrchestrationPathRouterImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: PagerDutyEventOrchestrationPathParent,
+				},
+			},
+			"sets": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1, // Router can only have 'start' set
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"rules": {
+							Type:     schema.TypeList,
+							Required: true, // even if there are no rules, API returns rules as an empty list
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"label": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"conditions": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: PagerDutyEventOrchestrationPathConditions,
+										},
+									},
+									"actions": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1, //there can only be one action for router
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"route_to": {
+													Type:     schema.TypeString,
+													Required: true,
+													ValidateFunc: func(v interface{}, key string) (warns []string, errs []error) {
+														value := v.(string)
+														if value == "unrouted" || value == "" {
+															errs = append(errs, fmt.Errorf("route_to within a set's rule has to be a service_id. Got: %q", v))
+														}
+														return
+													},
+												},
+											},
+										},
+									},
+									"disabled": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"catch_all": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actions": {
+							Type:     schema.TypeList,
+							Required: true,
```

Comment:
API does not accept payload with empty `actions`, So actions is made a required field. Since a rule at the minimum requires actions, passing rules as empty list for a set is not acceptable. 
Ref:
![image](https://user-images.githubusercontent.com/3372372/168080522-576aed4f-1c36-45aa-9366-5e8121db25ae.png)